open_instruct/vllm_utils3.py

```diff
@@ -190,55 +205,128 @@ def __init__(
+        while True:
+            if ray.get(self.actor_manager.should_stop.remote()):
+                self.logger.info("[LLMRayActor] Actor manager signaled to stop. Exiting generation loop.")
+                return
+            try:
+                request = self.prompt_queue.get(timeout=timeout)
+                outputs = self.llm.generate(
+                    sampling_params=request.sampling_params, prompt_token_ids=request.prompts, use_tqdm=False
+                )
+                result = self._process_outputs(outputs, dataset_index=request.dataset_index)
+                if request.is_eval:
+                    self.eval_results_queue.put(result)
+                else:
+                    self.results_queue.put(result)
+            except queue.Empty:
+                pass
+            except queue.Full:
+                self.logger.warning("Results queue is full, discarding result.")
+
+    def _run_generation_loop(self, timeout: float = 60.0):
+        """Run generation loop using LLMEngine directly."""
+        while True:
+            if ray.get(self.actor_manager.should_stop.remote()):
+                self.logger.info("[LLMRayActor] Actor manager signaled to stop. Exiting generation loop.")
+                return
+            try:
+                request = self.prompt_queue.get(timeout=timeout)
+            except queue.Empty:
+                self.logger.warning("[LLMRayActor] No request in the queue to process. Continuing.")
+                continue
+
+            # Process the request
+            prompts = request.prompts
+
+            # Debug logging
+            n_samples = request.sampling_params.n if request.sampling_params else 1
+            print(
+                f"[LLMRayActor] Processing batch: {len(prompts)} prompts, "
+                f"n={n_samples}, expecting {len(prompts) * n_samples} total outputs"
+            )
 
-    def generate(self, *args, **kwargs):
-        return self.llm.generate(*args, **kwargs)
+            # Add requests to the engine
+            for i, prompt in enumerate(prompts):
+                request_id = f"batch_{request.training_step}_{i}"
+                tokens_prompt = vllm.TokensPrompt(prompt_token_ids=prompt)
+                self.llm_engine.add_request(request_id, tokens_prompt, request.sampling_params)
+
+            # Run the engine event loop until all requests are finished
+            outputs = []
+            while self.llm_engine.has_unfinished_requests():
+                step_outputs = self.llm_engine.step()
+                for output in step_outputs:
+                    if output.finished:
+                        outputs.append(output)
+
+            # Sort outputs by prompt index (i)
+            # Request IDs are now in format: batch_{step}_{i}
+            outputs.sort(key=lambda x: int(x.request_id.split("_")[-1]))
+
+            # Debug logging
+            total_responses = sum(len(output.outputs) for output in outputs)
+            self.logger.info(
```

Comment:
Similar to past PRs, I wonder if we can make this logging explicitly debug level and not log by default, since it seems like a lot.

Reply:
I just removed this. Not worth including.